src/OMSimulatorLib/SystemTLM.cpp

```diff
@@ -262,13 +291,9 @@ void oms3::SystemTLM::disconnectFromSockets(const oms3::ComRef cref)
 
 oms_status_enu_t oms3::SystemTLM::setSocketData(const std::string &address, int managerPort, int monitorPort)
 {
-  omtlm_setAddress(model, address);
-  omtlm_setManagerPort(model, managerPort);
-  omtlm_setMonitorPort(model, monitorPort);
-
   this->address = address;
-  this->managerPort = managerPort;
-  this->monitorPort = monitorPort;
+  this->desiredManagerPort = managerPort;
```

Comment:
I think it would be better to check the model state and reject calls to `oms3::SystemTLM::setSocketData` if the model is instantiated, initialized or in simulation mode.

Reply:
That's true. Instantiated doesn't really matter, since socket connections are not established until initialization.